src/worker.rs

```diff
@@ -149,7 +151,19 @@ impl Worker {
                         } else {
                             None
                         };
-                        let res = req.resp_chan.send(Some(resp));
+                        let res;
+                        if mutated && !policy_evaluator.policy.mutating {
+                            let error_msg = "Non mutating policy tried to mutate a request. The request has been rejected.".to_string();
```

Comment:
This is the message that is shown to the final user. We should report something better/less scary.

> Request rejected by policy <policy name>. The policy attempted to mutate the request, but it is currently configured to not allow mutations.

Moreover, I would also emit a trace with what the mutation would look like. This can be useful for debugging purposes.

Reply:
+1!